### PR TITLE
Reduces memory requirements in JSON parser and adds bytes/s and peak memory usage to benchmarks

### PR DESCRIPTION
**Pull request by elstehle**
## Description
This PR reduces memory requirements in the new nested JSON parser and adds `bytes_per_second` and `peak_memory_usage` usage to benchmarks

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/benchmarks/io/json/nested_json.cpp [hunk 1]
```diff
@@ -71,15 +72,21 @@ void BM_NESTED_JSON(nvbench::state& state)
   state.add_element_count(input.size());
 
   // Run algorithm
+  auto mem_stats_logger = cudf::memory_stats_logger();
```
Comment (PointKernel) [inline cpp/benchmarks/io/json/nested_json.cpp:75]:
```suggestion
  auto const mem_stats_logger = cudf::memory_stats_logger();
```

File: cpp/src/io/json/nested_json_gpu.cu [hunk 2]
```diff
@@ -1117,6 +1116,9 @@ std::pair<rmm::device_uvector<PdaTokenT>, rmm::device_uvector<SymbolOffsetT>> ge
                                        stream};
 
   // Perform a PDA-transducer pass
+  rmm::device_scalar<SymbolOffsetT> num_written_tokens{stream, mr};
```
Comment (PointKernel) [inline cpp/src/io/json/nested_json_gpu.cu:1119]:
```suggestion
  rmm::device_scalar<SymbolOffsetT> num_written_tokens{stream};
```
temporary variables shouldn't use `mr`.

**Review by PointKernel — COMMENTED** [review]
Looks good with two nitpicks.

**Review by upsj — APPROVED** [review]

File: cpp/src/io/json/nested_json_gpu.cu (hunk 2 above)
Comment (elstehle) [reply cpp/src/io/json/nested_json_gpu.cu]: Thanks for pointing it out. Fixed 👍

**Review by karthikeyann — APPROVED** [review]
LGTM 👍

**Review by PointKernel — APPROVED** [review]

**Comment by elstehle** [comment]
@gpucibot merge

**Pull request merged**